math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -149,6 +159,12 @@ public void addVertexIfNotPresent(int v) {
         notifyListener();
     }
 
+    @Override
+    public boolean vertexExists(int v) {
+        checkInvalidVertex(v);
+        return v < vertices.size() && vertices.get(v) != null;
+    }
```

Comment:
I think you should not change the existing code: it's not necessary to extract the `checkInvalidVertex` in a method, consider duplicate the test here